src/consensus/aft/async_execution.h

```diff
@@ -12,8 +12,8 @@ namespace aft
     virtual void recv_append_entries(
       AppendEntries r, const uint8_t* data, size_t size) = 0;
     virtual void recv_append_entries_response(AppendEntriesResponse r) = 0;
-    virtual void recv_append_entries_signed_response(
-      SignedAppendEntriesResponse r) = 0;
+    virtual bool recv_append_entries_signed_response(
+      SignedAppendEntriesResponse r, bool is_pre_exec) = 0;
```

Comment:
I don't think I understand what the flag does, recv_append_entries_signed_response() is the consensus behaviour on receiving a signed response (ack/nack) to an append entries, but what does pre-execution mean in this context? Hasn't the request always been executed on the primary by that point?

Reply:
Sorry, I still don't understand what pre_exec means here, and there's no comment either, could it be clarified for my benefit please?

Reply:
To be split as two void calls as discussed.